### PR TITLE
Support `darwin/arm64` target

### PR DESCRIPTION
**Pull request by michaellzc**
- Add apple silicon support
- Fix buildkite pipeline
- Upgrade goreleaser to the latest 1.x release (there's no darwin_arm built for pre 1.x goreleaser)

### Test plan

🤞 

it passed
```sh
goreleaser release --snapshot --rm-dist
```

### CODE REVIEW

**Review by efritz — APPROVED** [review]

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+golang 1.18.1
```
Comment (Strum355) [inline .tool-versions:1]: Is this right, given we seem to be using 1.16 in release.yml? 
which also makes me wonder, is 1.16 right? :P @tjdevries @efritz @chrismwendt

Comment (michaellzc) [reply .tool-versions:1]: should we just bump `go-version` in release.yml to 1.18 as well?

Comment (Strum355) [reply .tool-versions:1]: That version has implications on lsif-go itself afaik

Comment (michaellzc) [reply .tool-versions]: @Strum355 reverted back to 1.16 for buildkite pipeline and it also supports m1 chip ;)

**Review by Strum355 — APPROVED** [review]
:nice:

**Comment by varungandhi-src** [comment]
Meta question: Why do we have both buildkite and GitHub Actions in this repo? Could we do with just one instead of both?

**Pull request merged**